File: Application/src/main/java/com/teeh/klimasensor/common/utils/CalcUtil.java
package com.teeh.klimasensor.common.utils;

import java.util.Collection;

/**
 * Created by teeh on 16.07.2017.
 */

public class CalcUtil {

    public static <T extends Number> Double sum(Collection<T> collection) {
        Double sum = 0.0;

        for (T entry : collection) {
            sum = sum + (Double)entry;
        }

        return sum;
    }

}

File: Application/src/main/java/com/teeh/klimasensor/common/utils/CalcUtil.kt
package com.teeh.klimasensor.common.utils

import java.util.ArrayList

object CalcUtil {

    fun <T : Number?> sum(collection: Collection<T>): Double {
        var sum: Double = 0.0

        for (entry in collection) {
            sum = sum + entry as Double
        }

        return sum
    }

    fun normalize(collection: Collection<Double>, normalization: Double?): List<Double> {
        val result = ArrayList<Double>()
        for (entry in collection) {
            result.add(entry / normalization!!)
        }
        return result
    }

}
